### PR TITLE
fix for "scroll to page anchor"

### PR DESCRIPTION
**Pull request by WesNetmo**
...and starting an attempt to clean this file.

### CODE REVIEW

**Comment by morrisonlevi** [comment]
Thanks for looking into this. I am not sure about some of the details; the `callbackCalled = true;` looks shady. I'm also not completely sure that this will prevent the browser's default scroll in some cases. I'll try and find some time to play with this to see how it works and if I can it to break.

**Comment by WesNetmo** [comment]
yes, because i don't know why $('html, body') was used. maybe for compatibility with some old browser. that causes the callback to be executed twice which is not what we want so for now i just prevented that to happen. clearly it's a temporary solution till i can investigate more (it's not different from previous implementation anyway)

File: js/common.js [hunk 1]
```diff
@@ -376,19 +410,7 @@ $(document).ready(function() {
         var $this = $(this);
         $this.append("<a class='genanchor' href='#" + $this.attr('id') + "'> ¶</a>");
     });
-
-    // Bind events for #[id]
-    $(window).on('hashchange', function(e) {
-      e.preventDefault();
```
Comment (WesNetmo) [inline js/common.js]: I've removed this because the hashchange event is not cancelable. It couldn't have ever worked

File: js/common.js [hunk 2]
```diff
@@ -376,19 +410,7 @@ $(document).ready(function() {
...
-      return false;
-    });
-
-    $('a[href^=#]').click(function(e) {
```
Comment (WesNetmo) [inline js/common.js]: also this was pretty much pointless, since `a[href^=#]` matches `<a href="#foo">` but not `<a href="foo.php#foo">`

**Pull request merged**